src/components/ec-checkbox/ec-checkbox.vue

```diff
@@ -0,0 +1,212 @@
+<template>
+  <div class="ec-checkbox">
+    <input
+      :id="id"
+      v-model="inputModel"
+      :aria-describedby="errorId"
+      type="checkbox"
+      class="ec-checkbox__input"
+      :disabled="disabled"
+    >
+    <label
+      :for="id"
+      class="ec-checkbox__label"
+      :class="{
+        'ec-checkbox__label--has-error': isInvalid || $slots['error-message'],
+        'ec-checkbox__label--large': large
+      }"
+    >
+      <span v-if="labelMessage || $slots['label-message']">
+        {{ labelMessage }}
+        <slot name="label-message" />
+      </span>
+    </label>
+
+    <div
+      :id="errorId"
+      v-if="isInvalid || $slots['error-message']"
+      class="ec-checkbox__error-text"
+    >
+      {{ errorMessage }}
+      <slot name="error-message" />
+    </div>
+  </div>
+</template>
+
+<script>
+export default {
+  name: 'EcCheckbox',
+  model: {
+    prop: 'checked',
+    event: 'checked-value-change',
+  },
+  props: {
+    checked: {
+      type: Boolean,
+    },
+    labelMessage: {
+      default: '',
+      type: String,
+    },
+    errorMessage: {
+      default: '',
+      type: String,
+    },
+    large: {
+      default: false,
+      type: Boolean,
+    },
+    disabled: {
+      default: false,
+      type: Boolean,
+    },
+  },
+  computed: {
+    id() {
+      return `ec-checkbox-${this._uid}`;
+    },
+    errorId() {
+      return this.isInvalid ? `ec-checkbox-error-${this._uid}` : null;
+    },
+    isInvalid() {
+      return !!this.errorMessage || !!this.$slots['error-message'];
+    },
+    inputModel: {
+      get() {
+        return this.checked;
+      },
+      set(checked) {
+        this.$emit('checked-value-change', checked);
+      },
+    },
+  },
+};
+</script>
+
+<style lang="scss">
+@import '../../scss/settings/index';
+@import '../../scss/tools/index';
+
+$ec-checkbox-white: $white !default;
+$ec-checkbox-tech-blue: $level-4-tech-blue !default;
```

Comment:
These variables make no sense, give them a meaning, is it background or hover or error or what color?. Remapping tech-blue to checkbox-tech-blue is waste of time, you can use the tech-blue in the rules directly.